util/util.go

```diff
@@ -1,9 +1,37 @@
 package util
 
-import "os/exec"
+import (
+	"errors"
+	"net"
+	"os/exec"
+	"strings"
+)
 
 // IsCommandAvailable check command exits.
 func IsCommandAvailable(cmd string) bool {
 	_, err := exec.LookPath(cmd)
 	return err == nil
 }
+
+func GetClientIp() (string, error) {
+	addresses, err := net.InterfaceAddrs()
+
+	if err != nil {
+		return "", err
+	}
+
+	for _, address := range addresses {
+		if ipnet, ok := address.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
+			if ipnet.IP.To4() != nil {
+				return ipnet.IP.String(), nil
+			}
+
+		}
+	}
+
+	return "", errors.New("can not find the client ip addresses")
+}
+
+func CountWords(s string) int {
+	return len(strings.Fields(s))
+}
```

Comment:
Overall, the code patch looks relatively safe and straightforward. Here are some suggestions for improvement:

1. In the `IsCommandAvailable` function, it might be helpful to include some logging or error messaging to indicate which command was not found if `err != nil`. This could help with debugging if the function fails for some reason.

2. The `GetClientIp` function could potentially be made more robust by checking for the presence of multiple IP addresses and returning the first non-loopback address found. This would help ensure that the function always returns a valid IP address.

3. In the `CountWords` function, it might be worth considering whether there are any edge cases where the `strings.Fields` function might not behave as expected. For example, it might not correctly handle strings with non-standard whitespace characters. One potential improvement would be to write a custom function that uses regular expressions to split the string into words.